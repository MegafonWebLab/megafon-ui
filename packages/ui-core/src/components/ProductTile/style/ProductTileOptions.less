@import '~styles/base.less';

@b: e('.mfui-product-tile-options');

@{b} {
    margin-left: 11px;
    margin-top: 16px;

    &__option {
        margin-top: 16px;
        display: flex;
    }
    &:not(:last-child) {
        margin-bottom: 8px;
    }

    &__head {
        margin-bottom: 8px;
    }

    &__icon {
        margin-right: 8px;
        svg,
        img {
            width: 32px;
            height: 32px;
        }
    }

    &__content {
        flex-shrink: 1;
    }

    &__title {
        .h5();

        a {
            .default-link();
            word-break: break-word;
        }
    }

    &__description {
        font-size: 12px;
        line-height: 16px;
    }

    &__cashback-hint {
        margin-top: -5px;
    }

    &__hint-text {
        font-weight: 300;
        p {
            padding: 0;
            margin-top: 0;
            margin-bottom: 10px;
        }

        a {
            border-bottom: 1px dashed;
            margin-right: 4px;

            &:hover {
                text-decoration: none;
                border: none;
            }
        }
    }

    &__bubble-trigger {
        display: inline-block;
        font-size: 12px;
        line-height: 16px;
        color: @blue;
        border-bottom: 1px dashed;
        font-weight: 400;
        margin: 0 4px 1px 2px;

        &:hover {
            border-bottom: none;
        }
    }
}
