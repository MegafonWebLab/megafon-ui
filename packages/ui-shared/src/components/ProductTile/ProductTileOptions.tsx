import * as React from 'react';
import * as PropTypes from 'prop-types';
import { cnCreate, BubbleHint } from '@megafon/ui-core';
import './style/ProductTileOptions.less';
import { IOption } from './ProductTile';

interface IProductTileOptionsProps {
    /** Has head */
    head?: string;
    /** Options */
    options: IOption[];
    /** Handle bubble */
    onClickBubble?: () => void;
}

const cn = cnCreate('mfui-product-tile-options');
class ProductTileOptions extends React.Component<IProductTileOptionsProps> {
    static propTypes = {
        head: PropTypes.string,
        options: PropTypes.arrayOf(PropTypes.shape({
            title: PropTypes.string,
            caption: PropTypes.string,
            value: PropTypes.string,
            unit: PropTypes.string,
            footnote: PropTypes.string,
            svgIcon: PropTypes.element,
        })),
        onClickBubble: PropTypes.func,
    };

    renderFootnote(title: string, footnote: string): JSX.Element {
        const { onClickBubble } = this.props;

        return (
            <div className={cn('content')}>
                <div className={cn('title')} dangerouslySetInnerHTML={{ __html: title }} />
                <BubbleHint
                    className={cn('cashback-c')}
                    popupWidth="small"
                    placement="bottom"
                    click
                    trigger={<span className={cn('bubble-trigger')} onClick={onClickBubble}>Подробнее</span>}
                >
                    <div
                        className={cn('hint-text')}
                        dangerouslySetInnerHTML={{ __html: footnote }}
                    />
                </BubbleHint>
            </div>
        );
    }

    renderContent(title: string, caption: string = '', value: string = '', unit: string = ''): JSX.Element {
        return (
            <div className={cn('content')}>
                <div className={cn('title')} dangerouslySetInnerHTML={{ __html: title }} />
                <div className={cn('description')}>
                    {caption || `${value} ${unit}`}
                </div>
            </div>
        );
    }

    renderIcon(svgIcon: JSX.Element | null = null): JSX.Element {
        return (
            <div className={cn('icon')}>{svgIcon}</div>
        );
    }

    render() {
        const { options, head } = this.props;

        return (
            <div className={cn('')}>
                {!!head && <div className={cn('head')}>{head}</div>}
                <div className={cn('wrapper')}>
                    {options.map((option, index) => {
                        const { title, caption, value, unit, svgIcon, footnote } = option;

                        return (
                            <div className={cn('option')} key={title + index}>
                                {this.renderIcon(svgIcon)}
                                {footnote
                                    ? this.renderFootnote(title, footnote)
                                    : this.renderContent(title, caption, value, unit)
                                }
                            </div>
                        );
                    })}
                </div>
            </div>
        );
    }
}

export default ProductTileOptions;
