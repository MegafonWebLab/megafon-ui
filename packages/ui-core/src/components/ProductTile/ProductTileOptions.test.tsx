import * as React from 'react';
import { shallow } from 'enzyme';
import ProductTileOptions from './ProductTileOptions';
import MegafonTv from 'icons/Service-logos/24/MF-TV.svg';

const props = {
    head: 'sdfsdfsdf',
    options: [{
        title: 'title',
        caption: 'caption',
        value: '234',
        unit: 'unit',
        footnote: 'footnote',
        svgIcon: <MegafonTv />,
    }],
};

describe('<ProductTileOptions />', () => {
    it('it renders ProductTileOptions', () => {
        const wrapper = shallow(<ProductTileOptions {...props} />);
        expect(wrapper).toMatchSnapshot();
    });
});
