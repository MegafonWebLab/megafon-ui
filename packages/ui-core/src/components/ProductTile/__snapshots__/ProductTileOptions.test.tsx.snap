// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ProductTileOptions /> it renders ProductTileOptions 1`] = `
<div
  className="mfui-product-tile-options"
>
  <div
    className="mfui-product-tile-options__head"
  >
    sdfsdfsdf
  </div>
  <div
    className="mfui-product-tile-options__wrapper"
  >
    <div
      className="mfui-product-tile-options__option"
      key="title0"
    >
      <div
        className="mfui-product-tile-options__icon"
      >
        <test-file-stub />
      </div>
      <div
        className="mfui-product-tile-options__content"
      >
        <div
          className="mfui-product-tile-options__title"
          dangerouslySetInnerHTML={
            Object {
              "__html": "title",
            }
          }
        />
        <BubbleHint
          className="mfui-product-tile-options__cashback-c"
          click={true}
          placement="bottom"
          popupAlign="left"
          popupPadding="normal"
          popupWidth="small"
          trigger={
            <span
              className="mfui-product-tile-options__bubble-trigger"
            >
              Подробнее
            </span>
          }
        >
          <div
            className="mfui-product-tile-options__hint-text"
            dangerouslySetInnerHTML={
              Object {
                "__html": "footnote",
              }
            }
          />
        </BubbleHint>
      </div>
    </div>
  </div>
</div>
`;
